File: Data Structures & Algorithms/house-robber/submission-0.cpp
class Solution {
   public:
    int rob(vector<int>& nums) {
        int prev2 = 0, prev1 = 0;
        int current_max = 0;
        for (const int& num : nums) {
            current_max = max(num + prev2, prev1);
            prev2 = prev1;
            prev1 = current_max;
        }

        return prev1;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        st = set()
        for i in nums:
            st.add(i)

        maxL = 0
        for i in nums:
            curr = i + 1
            currL = 1
            while curr in st:
                curr = curr + 1
                currL += 1
            maxL = max(currL, maxL)
        return maxL

            